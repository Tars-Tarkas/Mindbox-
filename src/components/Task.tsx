import React from "react";
import type { IToDo } from "../types/todo";

import { Checkbox } from "antd";

type Props = {
  todo: IToDo;
  updateToDo: (id: number) => void;
};

export default function Task({ todo, updateToDo }: Props) {
  return (
    <li
      style={{
        backgroundColor: "#fff",
        padding: "10px",
        marginBottom: "10px",
        listStyle: "none",
      }}
    >
      <Checkbox
        onChange={() => updateToDo(todo.id)}
        checked={todo.completed}
        name="ckeckbox"
        id={todo.text}
      >
        {todo.text}
      </Checkbox>
    </li>
  );
}
