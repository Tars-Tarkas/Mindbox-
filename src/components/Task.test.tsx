import React, { useContext } from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import type { IToDo } from "../types/todo";
import Task from "./Task";

const mockData = {
  id: 3,
  text: "ReactJS",
  completed: false,
} as IToDo;

describe("Task component", () => {
  const updateToDo = jest.fn();
  it("Task render", () => {
    render(<Task todo={mockData} updateToDo={updateToDo} key={mockData.id} />);
  });
  it("should show text of todos", () => {
    render(<Task todo={mockData} updateToDo={updateToDo} key={mockData.id} />);
    expect(screen.getByText(mockData.text)).toBeInTheDocument();
  });
  it("Task snapshot", () => {
    const task = render(
      <Task todo={mockData} updateToDo={updateToDo} key={mockData.id} />
    );
    expect(task).toMatchSnapshot();
  });
  it("Checkbox not checked", () => {
    render(<Task todo={mockData} updateToDo={updateToDo} key={mockData.id} />);
    const checkbox = screen.getByRole("checkbox");
    expect(checkbox).not.toBeChecked();
  });
});
