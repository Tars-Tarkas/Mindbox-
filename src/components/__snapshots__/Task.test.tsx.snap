// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component Task snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        style="background-color: rgb(255, 255, 255); padding: 10px; margin-bottom: 10px; list-style: none;"
      >
        <label
          class="ant-checkbox-wrapper css-dev-only-do-not-override-1hpnbz2"
        >
          <span
            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1hpnbz2"
          >
            <input
              class="ant-checkbox-input"
              id="ReactJS"
              name="ckeckbox"
              type="checkbox"
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            ReactJS
          </span>
        </label>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      style="background-color: rgb(255, 255, 255); padding: 10px; margin-bottom: 10px; list-style: none;"
    >
      <label
        class="ant-checkbox-wrapper css-dev-only-do-not-override-1hpnbz2"
      >
        <span
          class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1hpnbz2"
        >
          <input
            class="ant-checkbox-input"
            id="ReactJS"
            name="ckeckbox"
            type="checkbox"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
        <span>
          ReactJS
        </span>
      </label>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
